src/main/java/org/folio/auth/authtokenmodule/storage/ApiTokenStore.java

```diff
@@ -0,0 +1,187 @@
+   */
+  public Future<Void> createTableIfNotExists() {
+    // API tokens don't have an owning user. They are associated with a tenant
+    // only. The token itself is persisted since it will need to be viewed by
+    // end-users who have permission to see api tokens.
+    String createTable = "CREATE TABLE IF NOT EXISTS " +
+        tableName(tenant, API_TOKEN_SUFFIX) +
+        "(id UUID PRIMARY key, token TEXT NOT NULL, " +
+        "is_revoked BOOLEAN NOT NULL, issued_at INT8 NOT NULL)";
+
+    log.info("Creating {} tables", ApiTokenStore.class.getName());
+
+    return pool.query(createTable).execute().mapEmpty();
+  }
+
+  /**
+   * Save the token to the token store. This should be done anytime a new API token is
+   * issued.
+   * @param conn An SqlConnection object to be used by the method to access the token
+   * store. It is the responsibility of callers to close this connection.
+   * @param apiToken The API token to store.
+   * @return A failed future should the save operation fail. Otherwise a succeeded future
+   * is returned.
+   */
+  public Future<Void> saveToken(ApiToken apiToken) {
+    UUID id = apiToken.getId();
+    long issuedAt = apiToken.getIssuedAt();
+    boolean isRevoked = false;
+    String tenant = apiToken.getTenant();
+
+    String token = "";
+    try {
+      token = apiToken.encodeAsJWT(tokenCreator);
+    } catch (Exception e) {
+      log.error("Unable to encode token when saving: {}", e.getMessage());
+      return Future.failedFuture("Unable to encode token when saving: " + e.getMessage());
+    }
+
+    log.info("Inserting token id {} into {} token store", id, API_TOKEN_SUFFIX);
+
+    String insert = "INSERT INTO " + tableName(tenant, API_TOKEN_SUFFIX) +
+        "(id, token, is_revoked, issued_at) VALUES ($1, $2, $3, $4)";
+    var values = Tuple.of(id, token, isRevoked, issuedAt);
+
+    return pool.preparedQuery(insert).execute(values).mapEmpty();
+
+    // TODO Should we return the encoded API token to callers?
+  }
+
+  /**
+   * Check that the token has not been revoked. This will return a failed future if
+   * the token has been revoked, otherwise it will return a succeeded future.
+   * @param refreshToken The API token to check.
+   * @return A failed future if the token has been revoked. Otherwise a succeeded future
+   * is returned.
+   */
+  public Future<Void> checkTokenNotRevoked(ApiToken apiToken) {
+    UUID tokenId = apiToken.getId();
+
+    log.info("Checking revoked status of {} api token id {}",
+      API_TOKEN_SUFFIX, tokenId);
+
+    String select = "SELECT is_revoked FROM " + tableName(tenant, API_TOKEN_SUFFIX) +
+      "WHERE id=$1";
+    Tuple where = Tuple.of(tokenId);
+
+    return pool.withConnection(conn -> {
+      return getRow(conn, select, where).compose(row -> {
+        Boolean isRevoked = row.getBoolean("is_revoked");
+
+        log.info("Revoked status of {} token id {} is {}", API_TOKEN_SUFFIX, tokenId, isRevoked);
```

Comment:
Log messages slow down the software. Can you go through your pull request and reduce the log level from info to debug for log messages that are not needed for production? No info message is needed at all if the token is not revoked because other log messages show that the request to mod-authtoken is successful.

Reply:
Agree. Do you have an opinion here on what _should_ be useful info messages? I'm thinking perhaps:
* When a RT token has been detected as leaked and revoked
* When an ApiToken has been revoked